[#21] TestContainer 적용

dvlprkoji opened the pull request:
[#21] TestContainer 적용

dvlprkoji committed:
refactor: CouponStock 관련 메소드 EventService 에서 couponStockService 로 이동

dvlprkoji committed:
refactor: CommonResponse 클래스 패키지 이동

dvlprkoji committed:
refactor: @Value 대신 @ConfigurationProperties 사용

dvlprkoji committed:
refactor: CouponStock 관련 메소드 EventService 에서 couponStockService 로 이동

dvlprkoji committed:
feat: testcontainers 라이브러리 적용

dvlprkoji committed:
test: admin-api 모듈 EventControllerTest 추가

- TestContainer 적용

dvlprkoji committed:
refactor: JPA 테이블 키 생성전략 변경 (AUTO > IDENTITY)

dvlprkoji committed:
refactor: EventJoinMember 변경

- Redis 자료구조 변경 (Strings > Sets)
- key 값 변경 ("event:{eventId}:date:{eventDate}:member:{memberId}" > "{eventId}:{eventDate}")

dvlprkoji committed:
test: coupon-api 모듈 EventControllerTest 추가

dvlprkoji committed:
refactor: @Value 대신 @ConfigurationProperties 사용

dvlprkoji committed:
fix: Profile 명 수정 (dev > !test)

dvlprkoji committed:
refactor: @Value 대신 @ConfigurationProperties 사용

dvlprkoji committed:
fix: 테스트 초기정보 @BeforeEach 에서 주입하도록 수정

dvlprkoji committed:
refactor: Kafka message key 로 memberId 를 설정

dvlprkoji committed:
fix: Kafka Producer 설정 변경

- 브로커 1개 > 3개 변경

dvlprkoji committed:
refactor: EventJoinDTO 클래스 패키지 이동

- aladiin.couponapi.model.dto > aladiin.core.dto

dvlprkoji committed:
feat: 이벤트 참여 Kafka Listener 추가

dvlprkoji committed:
feat: 이벤트 중복 참여 판단 로직 추가

f-lab-lyan commented on admin-api/src/test/java/aladiin/adminapi/config/RedisConfiguration.java:
기본 configuration과 합쳐도 되지 않을까요?

f-lab-lyan commented on aladiin-core/src/main/java/aladiin/core/domain/entity/dao/EventJoinRepository.java:
```diff
@@ -10,7 +10,7 @@ public class EventJoinRepository {
-    public boolean findMember(String key) {
-        return redissonClient.getBucket(key).isExists();
+    public boolean findMember(String key, Long value) {
+        return redissonClient.getSet(key).contains(value);
```
```suggestion
        return redissonClient.exists(key + '{' + value + '}');
```

dvlprkoji committed:
refactor: EventJoinMember 변경

- Redis 자료구조 변경 (Sets > Strings)
- key 값 변경 ("{eventId}:{eventDate}" > "{eventId}:{eventDate}:{memberId}")

dvlprkoji committed:
test: HTTP 요청 클라이언트 변경 (RestClient > TestRestTemplate)

dvlprkoji committed:
refactor: 패키지 이동

- aladiin.couponapi.model.dto.EventJoinRequest > aladiin.core.request.EventJoinRequest;
- com.aladiin.couponconsumer > aladiin.couponconsumer

dvlprkoji committed:
test: EventServiceTest 추가

f-lab-lyan commented on coupon-consumer/src/main/java/aladiin/couponconsumer/service/KafkaMessageListener.java:
```diff
@@ -0,0 +1,42 @@
+        // 3. 쿠폰 발급
+        //  ㄴ 메모리 쿠폰 재고 감소
+        //  ㄴ RDB 쿠폰 발급 처리
+        couponService.issue(eventJoinRequest.getMemberId());
```
https://ko.wikipedia.org/wiki/%EB%A9%B1%EB%93%B1%EB%B2%95%EC%B9%99
